diff -urN resample-1.8.1/sndlib-20/audio.c resample-1.8.1-patched/sndlib-20/audio.c
--- resample-1.8.1/sndlib-20/audio.c	2006-11-11 08:03:19.000000000 +1000
+++ resample-1.8.1-patched/sndlib-20/audio.c	2012-11-24 01:21:59.616253524 +1100
@@ -2013,7 +2013,7 @@
           if (ioctl(audio_out, SNDCTL_DSP_SETFRAGMENT, &buffer_info) == -1) 
 	    {
 	      char *tmp;
-	      fprintf(stderr, tmp = mus_format("can't set %s fragments to: %d x %d",
+	      fprintf(stderr, "%s\n", tmp = mus_format("can't set %s fragments to: %d x %d",
 					       dev_name, FRAGMENTS, FRAGMENT_SIZE)); /* not an error if ALSA OSS-emulation */
 	      fragment_set_failed = true;
 	      FREE(tmp);
@@ -9747,7 +9747,7 @@
     {
       if ((print_it) || (!(save_it)))
 	{
-	  mus_print(str);
+	  mus_print("%s\n",str);
 	}
       else
 	{
diff -urN resample-1.8.1/sndlib-20/headers.c resample-1.8.1-patched/sndlib-20/headers.c
--- resample-1.8.1/sndlib-20/headers.c	2006-11-11 08:03:19.000000000 +1000
+++ resample-1.8.1-patched/sndlib-20/headers.c	2012-11-24 01:24:39.366532797 +1100
@@ -5097,7 +5097,7 @@
   mus_error_set_handler(old_error_handler);
   CLOSE(chan, name);
   if (err != MUS_NO_ERROR)
-    return(mus_error(local_error_type, local_error_msg)); /* pass error info on up the chain now that we've cleaned up the open file descriptor */
+    return(mus_error(local_error_type, "%s", local_error_msg)); /* pass error info on up the chain now that we've cleaned up the open file descriptor */
   return(err);
 }
 
diff -urN resample-1.8.1/sndlib-20/sound.c resample-1.8.1-patched/sndlib-20/sound.c
--- resample-1.8.1/sndlib-20/sound.c	2006-11-11 08:03:19.000000000 +1000
+++ resample-1.8.1-patched/sndlib-20/sound.c	2012-11-24 01:23:19.564393751 +1100
@@ -60,7 +60,7 @@
     (*mus_error_handler)(error, mus_error_buffer);
   else 
     {
-      fprintf(stderr, mus_error_buffer);
+      fprintf(stderr, "%s", mus_error_buffer);
       fputc('\n', stderr);
     }
   return(MUS_ERROR);
@@ -176,7 +176,7 @@
 static void default_mus_error(int ignore, char *msg)
 {
   /* default error handler simply prints the error message */
-  fprintf(stderr, msg);
+  fprintf(stderr, "%s", msg);
 }
 
 static time_t local_file_write_date(const char *filename)
@@ -1031,7 +1031,7 @@
   char *errmsg;
   errmsg = mus_array_to_file_with_error(filename, ddata, len, srate, channels);
   if (errmsg)
-    return(mus_error(MUS_CANT_OPEN_FILE, errmsg));
+    return(mus_error(MUS_CANT_OPEN_FILE, "%s", errmsg));
   return(MUS_NO_ERROR);
 }
 
@@ -1067,6 +1067,6 @@
   FREE(idata);
 #endif
   if (errmsg)
-    return(mus_error(MUS_CANT_OPEN_FILE, errmsg));
+    return(mus_error(MUS_CANT_OPEN_FILE, "%s", errmsg));
   return(MUS_NO_ERROR);
 }
diff -urN resample-1.8.1/src/filterkit.c resample-1.8.1-patched/src/filterkit.c
--- resample-1.8.1/src/filterkit.c	2006-11-11 08:03:14.000000000 +1000
+++ resample-1.8.1-patched/src/filterkit.c	2012-11-24 01:21:59.616253524 +1100
@@ -559,7 +559,7 @@
       sprintf(s,"\n%s%s", prompt, (*help) ? " (Type ? for help)" : "");
       getstr(s,(deflt)?"yes":"no",s);
       if (*s=='?' && *help)
-         printf(help);
+         printf("%s\n",help);
       if (*s=='Y' || *s=='y')
          return(TRUE);
       if (*s=='N' || *s=='n')
@@ -577,7 +577,7 @@
       sprintf(s,"\n%s%s",prompt, (*help) ? " (Type ? for Help)" : "");
       getstr(s,deflt,s);
       if (*s=='?' && *help)
-         printf(help);
+         printf("%s\n",help);
       else
          return(s);
       }
@@ -595,7 +595,7 @@
       sprintf(sdeflt,"%g",deflt);
       getstr(s,sdeflt,s);
       if (*s=='?' && *help)
-         printf(help);
+         printf("%s\n",help);
       else
          {
          if (!sscanf(s,"%lf",&newval))
@@ -617,7 +617,7 @@
       sprintf(sdeflt,"%d",deflt);
       getstr(s,sdeflt,s);
       if (*s=='?' && *help)
-         printf(help);
+         printf("%s\n",help);
       else
          {
          if (!sscanf(s,"%d",&newval))
diff -urN resample-1.8.1/src/resample.c resample-1.8.1-patched/src/resample.c
--- resample-1.8.1/src/resample.c	2006-11-11 08:03:13.000000000 +1000
+++ resample-1.8.1-patched/src/resample.c	2012-11-24 01:21:59.616753525 +1100
@@ -123,7 +123,7 @@
 	    knowFactor = TRUE;
 	    break;
 	case 'v':			       /* -version */
-	    printf(resampleVersion);
+	    printf("%s\n",resampleVersion);
 	    if (argc == 1)
 		exit(0);
 	    break;
